### PR TITLE
  do analyzer at nodes

### PR DESCRIPTION
**Pull request by Shaoshian**
Signed-off-by: shaoshian <1196753438@qq.com>

### CODE REVIEW

**Comment by Shaoshian** [comment]
@xuechendi   收到了么？ 我发的对么？？

File: benchmarking/mod/benchmark.py [hunk 1]
```diff
@@ -248,25 +322,98 @@ def archive(self):
         #write description to dir
         with open( "%s/conf/description" % dest_dir, 'w+' ) as f:
             f.write( self.benchmark["description"] )
+
+        #copy all.conf to all node's log path
+        self.cetune_path = os.path.join(self.cluster["tmp_dir"],'CeTune/')
+        self.loacl_cetune_path = '/'
```
Comment (xuechendi) [inline benchmarking/mod/benchmark.py]: loacl_cetune_path? local_cetune_path?

File: benchmarking/mod/benchmark.py [hunk 2]
```diff
@@ -59,10 +63,77 @@ def go(self, testcase, tuning):
...
         common.printout("LOG","Post Process Result Data")
+        total_result = self.combine_nodes_result()
+        common.printout("LOG","Write analyzed results into result.json")
         try:
-            analyzer.main(['--path', self.cluster["dest_dir"], 'process_data'])
+            with open('%s/result.json' % self.cluster["dest_dir"], 'w') as f:
+                json.dump(total_result, f, indent=4)
+            view = visualizer.Visualizer(total_result, self.cluster["dest_dir"])
+            output = view.generate_summary_page()
         except:
```
Comment (xuechendi) [inline benchmarking/mod/benchmark.py]: prefer to call analyzer here instead of visualizer

File: benchmarking/mod/benchmark.py [hunk 3]
```diff
@@ -59,10 +63,77 @@ def go(self, testcase, tuning):
...
-            common.printout("ERROR","analyzer failed, pls try cd analyzer; python analyzer.py --path %s process_data " % self.cluster["dest_dir"])
+             common.printout("ERROR","Write analyzed results into result.json failed")
+
+
+    def combine_nodes_result(self):
```
Comment (xuechendi) [inline benchmarking/mod/benchmark.py]: I don't think this function should existing in benchmark.py

File: benchmarking/mod/benchmark.py [hunk 4]
```diff
@@ -59,10 +63,77 @@ def go(self, testcase, tuning):
...
+            self.combine_result = ana.summary_result(self.combine_result)
+            self.combine_result["summary"]["Download"] = {"Configuration":{"URL":"<button class='cetune_config_button' href='../results/get_detail_zip?session_name=%s&detail_type=conf'><a>Click TO Download</a></button>" % self.combine_result["session_name"]}}
+            node_ceph_version = {}
+            if ana.collect_node_ceph_version(os.path.join(self.cluster["dest_dir"],'raw')):
+                for key,value in ana.collect_node_ceph_version(os.path.join(self.cluster["dest_dir"],"raw")).items():
+                    node_ceph_version[key] = {"ceph_version":value}
+            self.combine_result["summary"]["Node"] = node_ceph_version
+	return self.combine_result
+
+    def check_osd_result_exists(self):
```
Comment (xuechendi) [inline benchmarking/mod/benchmark.py]: move to analyzer?

File: benchmarking/mod/benchmark.py [hunk 5]
```diff
@@ -248,25 +322,98 @@ def archive(self):
...
+            for client in self.benchmark["distribution"].keys():
+                common.scp(user, client, "%s/conf/all.conf" % (self.loacl_cetune_path), "%s/all.conf" % self.cluster["tmp_dir"])
+        except Exception:
+            common.printout("ERROR","copy all.conf to nodes failed!")
+            pass
+
+        #do analyzer at node
+        abnormal_node = []
+        self.threads = []
+        self.thrad_id = {}
```
Comment (xuechendi) [inline benchmarking/mod/benchmark.py]: thrad_id? what is thrad?

File: benchmarking/mod/benchmark.py [hunk 6]
```diff
@@ -248,25 +322,98 @@ def archive(self):
...
+
             common.bash("mkdir -p %s/raw/%s" % (dest_dir, node))
             common.rscp(user, node, "%s/raw/%s/" % (dest_dir, node), "%s/*.txt" % self.cluster["tmp_dir"])
+            common.rscp(user, node, "%s/raw/%s/" % (dest_dir, node), "%s/*.json" % self.cluster["tmp_dir"])
             if "blktrace" in self.cluster["collector"]:
                 common.rscp(user, node, "%s/raw/%s/" % (dest_dir, node), "%s/*blktrace*" % self.cluster["tmp_dir"])
             if "lttng" in self.cluster["collector"]:
                 common.rscp(user, node, "%s/raw/%s/" % (dest_dir, node), "%s/lttng-traces" % self.cluster["tmp_dir"])
 
+        #do analyzer at client
```
Comment (xuechendi) [inline benchmarking/mod/benchmark.py]: don't do analyze inside benchmark

File: benchmarking/mod/benchmark.py (hunk 1 above)
Comment (Shaoshian) [inline benchmarking/mod/benchmark.py]: 这个就是head上面的cetune的位置，这个位置不需要强制放到/opt/下面  所以会根据code的运行环境来得到head上面的cetune的路径

File: benchmarking/mod/benchmark.py (hunk 5 above)
Comment (Shaoshian) [inline benchmarking/mod/benchmark.py]: thrad_id  这个代码改过后，用不到这个变量了，下次修改时去掉这个变量

File: benchmarking/mod/benchmark.py (hunk 1 above)
Comment (xuechendi) [inline benchmarking/mod/benchmark.py]: so I believe this is a spell error right? I don't understand the word 'loacl', what does this word mean?

Comment (Shaoshian) [inline benchmarking/mod/benchmark.py]: 恩  修改成head_cetune_path  可以不？

Comment (xuechendi) [inline benchmarking/mod/benchmark.py]: why don't name it as local_cetune_path? anyway, head_cetune_path also works for me

**Pull request closed without merging**